src/ethers/signer.ts

```diff
@@ -50,16 +52,84 @@ function restoreCustomFields(tx: Deferrable<TransactionRequest>, savedFields: an
   }
 }
 
-// Wallet API dynamic hooking
-// Wallet.populateTransaction = function() {
-// };
 
-export class KlaytnWallet extends Wallet {
+export class Wallet extends OldWallet {
 
-  checkTransaction(transaction: Deferrable<TransactionRequest>): Deferrable<TransactionRequest> {
+  private klaytn_address: string | undefined;
+  
+  private dynamicUpdateWalletAPI;
+  
+  _checkTransaction:((transaction: Deferrable<TransactionRequest>) => Deferrable<TransactionRequest>) | undefined ;
+  _populateTransaction: ((transaction: Deferrable<TransactionRequest>) => Promise<TransactionRequest>) | undefined ;
+
+  _signTransaction; 
+  _sendTransaction;
+
+  constructor(address: any, privateKey?: any, provider?: Provider, dynamicUpdateWalletAPI: boolean=true ) {
+    let str_addr = String(address); 
+
+    if ( HexStr.isHex(address) && str_addr.length == 42 && str_addr.startsWith("0x")) {
+      super( privateKey, provider); 
+      this.klaytn_address = address; 
+    } else if ( HexStr.isHex(address) && str_addr.length == 40 && !str_addr.startsWith("0x")) {
+      super( privateKey, provider); 
+      this.klaytn_address = "0x" + address; 
+    } else {
+      provider = privateKey; 
+      privateKey = address;
+      super( privateKey, provider); 
+    }
+
+    // KlaytnWallet API is also working on Wallet 
+    // For example, Wallet.populateTransaction is same with KlaytnWallet.populateTransaction. 
+    this.dynamicUpdateWalletAPI = dynamicUpdateWalletAPI;
+    if ( this.dynamicUpdateWalletAPI == true ) {
+      this._checkTransaction = super.checkTransaction; 
+      super.checkTransaction = this.checkTransaction; 
+
+      this._populateTransaction = super.populateTransaction; 
+      super.populateTransaction = this.populateTransaction; 
+
+      this._signTransaction = super.signTransaction;
+      super.signTransaction = this.signTransaction; 
+
+      // @ts-ignore
+      super.signTransactionAsFeePayer = this.signTransactionAsFeePayer;
 
+      this._sendTransaction = super.sendTransaction;
+      super.sendTransaction = this.sendTransaction; 
+
+      // @ts-ignore
+      super.sendTransactionAsFeePayer = this.sendTransactionAsFeePayer;
+    }
+  }
+
+  getAddress(): Promise<string> {
+    if ( this.klaytn_address == undefined ) 
+      return super.getAddress();
+    return Promise.resolve( String(this.klaytn_address) );
```

Comment:
I guess you can just `return this.klaytn_address` and compiler automatically makes it into promise?

Reply:
this.klaytn_address is string. 
So type script error occurs.